src/librustc_lint/types.rs

```diff
@@ -176,6 +177,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for TypeLimits {
                             _ => bug!(),
                         };
                         if lit_val < min || lit_val > max {
+                            let parent_id = cx.tcx.hir.get_parent_node(e.id);
+                            if let hir_map::NodeExpr(parent_expr) = cx.tcx.hir.get(parent_id) {
+                                if let hir::ExprCast(..) = parent_expr.node {
+                                    if let ty::TyChar = cx.tables.expr_ty(parent_expr).sty {
+                                        let mut err = cx.struct_span_lint(
+                                                             OVERFLOWING_LITERALS,
+                                                             e.span,
+                                                             "only u8 can be casted into char");
+                                        err.help(&format!("try with `'\\u{:X}' as char`", lit_val));
```

Comment:
Could this be a `span_suggestion` instead? Specially so that the help _doesn't_ appear _after_ the "lint defined here" span. Also, that way it can be easily applied by RLS enabled tools.

You don't need to cast (`as char`) when you have a literal `char` (`\uEF8888`).

I would probably word this as: "instead of casting this numeric literal, use a `char` literal".